### PR TITLE
feat(ui): better dragging in shops (extends shop ship selection changes)

### PR DESCRIPTION
**Pull request by flowers9**
**Feature:**

## Feature Details
This changes how dragging player ships works in shops (shipyard and outfitter).  Currently, dragging only moves one ship, and changes the position in the shiplist as the player drags it.  This change makes dragging drag all the selected ships, and doesn't change their position in the ship list until released (if released in the ship list, otherwise there is no change).

Also, the ship in the ship list with the mouse over it is highlighted, primarily to allow easier drag drop positioning since the dragged ship position is no longer being updated on the fly.

## UI Screenshots
Before:
Screenshot from 2023-07-25 17-02-43
Highlighting example:
Screenshot from 2023-07-25 17-02-16
Drag example:
Screenshot from 2023-07-25 17-08-21

## Testing Done
So much dragging was done.  Selling or buying during a drag (which cancels it).

## Performance Impact
The extra dragging code introduces minor overhead when dragging.  There's also a (probably trivial) check for hovering when drawing the ship list.  And two new (small) icons in memory.

## Notes
This builds on #9069, so it includes all that code, including changes to it.  It could be written without that, but I'm hoping 9069 makes it in.  And changes to 9069 will get reflected here, if need be.

### CODE REVIEW

File: images/ui/icon unselected highlighted.png
Comment (RisingLeaf) [inline images/ui/icon unselected highlighted.png:1]: copyright for those?
add it to the copyright file

File: source/OutfitterPanel.cpp [hunk 1]
```diff
@@ -371,15 +371,15 @@ ShopPanel::BuyResult OutfitterPanel::CanBuy(bool onlyOwned) const
 		if(isInStore)
 		{
 			// Player hit 'i' or 'c' when they should've hit 'b'.
-			if(playerShip)
+			if(shipSelection.Selected())
```
Comment (RisingLeaf) [inline source/OutfitterPanel.cpp:374]: you are calling the getter a few times inside this function, how about you store it in a variable?

File: source/ShipSelection.cpp [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	{
+		Set(player.Flagship());
+		return nullptr;
+	}
+
+	// Find the currently selected ship in the list.
+	vector<shared_ptr<Ship>>::const_iterator it = player.Ships().begin();
+	for( ; it != player.Ships().end(); ++it)
+		if(it->get() == selectedShip)
+			break;
```
Comment (RisingLeaf) [inline source/ShipSelection.cpp:70]:
```suggestion
	vector<shared_ptr<Ship>>::const_iterator it = player.Ships().find(selectedShip);
```
wouldn't this be the same? (dunno if the function is like that or `find(player.Ships.begin(), player.Ships().end(), slectedShip);`)

**Review by RisingLeaf — CHANGES_REQUESTED** [review]
haven't reviewed everything yet but these are the things I noted so far

File: source/ShipSelection.cpp (hunk 2 above)
Comment (flowers9) [reply source/ShipSelection.cpp:70]: Well, player.Ships() is a vector, which doesn't have a find method.  Also, selectedShip is a Ship *, not a shared_ptr<Ship>.  But I guess I could figure out how to use find with a lamda.

Comment (RisingLeaf) [reply source/ShipSelection.cpp:70]: a lambda would be nice yes,

**Comment by quyykk** [comment]
One change this PR does is if you drag a ship to a position that comes after its previous position, the ship is positioned one place before the actual drag position. 

Here I'm dragging the second ship to the third position:

image

but then when I release it doesn't change

**Comment by quyykk** [comment]
The same behavior still occurs when dragging multiple ships (sorry :sweat_smile:)

**Comment by flowers9** [comment]
Well, first, see above comment.  Second, it's not possible to always have the selected ship of a multi-ship drag and drop end up under the cursor.  At least, not without reordering the dragged ships themselves, which would not be intuitive.

So, the group of dragged ships will always get placed to the left/right of the dropped on ship, just like with a single dragged ship.  The order of the dragged ships (relative to themselves) will remain the same.  The order of the non-dragged ships will remain the same (modulo deleting and inserting the dragged ships).  That seems the best approach, to me.

**Comment by flowers9** [comment]
Eh, it's been long enough, I think.

**Pull request closed without merging**